src/main/java/com/gamingmesh/jobs/CMILib/CMIMaterial.java

```diff
@@ -2508,4 +2508,20 @@ public String getBukkitName() {
     public void setBukkitName(String bukkitName) {
 	this.bukkitName = bukkitName;
     }
+
+    public static String getGeneralMaterialName(String fullName) {
+        String newName = fullName.toUpperCase();
+        if (newName.startsWith("STRIPPED")) {
+            return newName.replaceFirst("_[^_]+", "");
+        }
+        else if (newName.matches("^(DARK|LIGHT).+")) {
+            return newName.replaceFirst(".+?_.+?_", "");
+        }
+        else if (newName.matches(
+                "^(WHITE|ORANGE|MAGENTA|YELLOW|LIME|PINK|GRAY|CYAN|PURPLE|BLUE|BROWN|GREEN|RED|BLACK|" +
+                "OAK|SPRUCE|BIRCH|JUNGLE|ACACIA).+")) {
```

Comment:
What's the result if we have `banner-all`, `leaves-all`... We should try to find a way to check whatever a material name have `-all` sub name.

Reply:
This should already work. If you have `banner-all`, it'll first internally be like white_banner or black_banner. So it is saved as white_banner-all and this matches with this if condition.
I did not dive deep into how that default coloring works when you just say banner-all, but it actually works. Did you test it?